### ISSUE TITLE
cmd/utils: `--jspath` doesn't expand home directory

### ISSUE THREAD

**Issue by lightclient** [ethereum/go-ethereum#22889]
Tilde not expanded:
```console
$ pwd
/home/lightclient/go-ethereum

$ ls ../scripts
file.js

$ geth attach --jspath=~/scripts --preload=file.js http://localhost:8545
Fatal: Failed to start the JavaScript console: file.js: open ~/scripts/file.js: no such file or directory
```

But absolute path works:
```
$ geth attach --jspath=/home/lightclient/scripts --preload=file.js http://localhost:8545
Welcome to the Geth JavaScript console!
...
```

Looks like `DocRoot` is not being expanded? https://github.com/ethereum/go-ethereum/blob/f19a679b09500e63b2fd134a4e829591174d9e48/cmd/geth/consolecmd.go#L90

Then it's used here: https://github.com/ethereum/go-ethereum/blob/d8ff53dfb8a516f47db37dbc7fd7ad18a1e8a125/internal/jsre/jsre.go#L242

edit: removed references to broken relative path since it was a user error :)

**Comment by holiman** [comment]
Relative paths works for me: 

```
[user@work go-ethereum]$ ls ../foo
deleteme.js
[user@work go-ethereum]$ geth --jspath=../foo --preload=deleteme.js  attach /tmp/geth.ipc
This import worked fine
```

**Comment by holiman** [comment]
Does this fix the `../scripts`-case for you? 
```diff
diff --git a/cmd/geth/consolecmd.go b/cmd/geth/consolecmd.go
index 85eabb527b..81e5a05537 100644
--- a/cmd/geth/consolecmd.go
+++ b/cmd/geth/consolecmd.go
@@ -85,9 +85,13 @@ func localConsole(ctx *cli.Context) error {
 	if err != nil {
 		utils.Fatalf("Failed to attach to the inproc geth: %v", err)
 	}
+	docRoot, err := filepath.Abs(ctx.GlobalString(utils.JSpathFlag.Name))
+	if err != nil {
+		utils.Fatalf("Failed to determine the js root %v", err)
+	}
 	config := console.Config{
 		DataDir: utils.MakeDataDir(ctx),
-		DocRoot: ctx.GlobalString(utils.JSpathFlag.Name),
+		DocRoot: docRoot,
 		Client:  client,
 		Preload: utils.MakeConsolePreloads(ctx),
 	}
@@ -144,9 +148,13 @@ func remoteConsole(ctx *cli.Context) error {
 	if err != nil {
 		utils.Fatalf("Unable to attach to remote geth: %v", err)
 	}
+	docRoot, err := filepath.Abs(ctx.GlobalString(utils.JSpathFlag.Name))
+	if err != nil {
+		utils.Fatalf("Failed to determine the js root %v", err)
+	}
 	config := console.Config{
 		DataDir: utils.MakeDataDir(ctx),
-		DocRoot: ctx.GlobalString(utils.JSpathFlag.Name),
+		DocRoot: docRoot,
 		Client:  client,
 		Preload: utils.MakeConsolePreloads(ctx),
 	}
@@ -197,9 +205,13 @@ func ephemeralConsole(ctx *cli.Context) error {
 	if err != nil {
 		utils.Fatalf("Failed to attach to the inproc geth: %v", err)
 	}
+	docRoot, err := filepath.Abs(ctx.GlobalString(utils.JSpathFlag.Name))
+	if err != nil {
+		utils.Fatalf("Failed to determine the js root %v", err)
+	}
 	config := console.Config{
 		DataDir: utils.MakeDataDir(ctx),
-		DocRoot: ctx.GlobalString(utils.JSpathFlag.Name),
+		DocRoot: docRoot,
 		Client:  client,
 		Preload: utils.MakeConsolePreloads(ctx),
 	}
```

**Comment by lightclient** [comment]
@holiman apologies - I ran into the issue when trying to do tilde expansion and while quickly testing access via a relative path I must've messed something up. I tried again just now and they were working. 

The tilde expansion still wasn't working though after applying your diff. I believe I fixed it in #22900.
